mandatory fields in signed ACF request objects (#285)
camaraproject/IdentityAndConsentManagement: documentation/CAMARA-Security-Interoperability.md

It is RECOMMENDED that signed authentication requests be used, as specified by [OIDC](https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests). The same key MAY be used for signing the authentication request as is used for [client authentication](#client-authentication). In addition, this document defines that:
  * Values for the mandatory parameters `scope`, `response_type`, `client_id` and `redirect_uri` MUST be included using the OAuth2.0 request syntax. The values for these parameters MUST match those in the signed request object.
  * The fields iss, aud, iat, exp and jti MUST be included in the signed request object.
  * For all other parameters, the authorization server MUST only use the parameters contained in the signed request object. 

>   * The request SHALL be rejected by the authorisation server if the exp claim is more than 300 seconds later than the time of receipt. Additionally, if the iat claim is present, the request SHALL be rejected if the difference between the exp claim and iat claim is more than 300 seconds.

Reviewer: @AxelNennker, I would add the same paragraph regarding the expiration date in the CIBA flow section.

Author: Maybe I am confused, this is the text from the CIBA section:
> The request SHALL be rejected by the authorisation server if the exp claim is more than 300 seconds later than the time of receipt. Additionally, if the iat claim is present, the request SHALL be rejected if the difference between the exp claim and iat claim is more than 300 seconds.

It is exactly the same.

Reviewer 2: @AxelNennker The text quoted above is actually from the [Client Authentication](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#client-authentication) section, not the [CIBA](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#client-initiated-backchannel-authentication-flow) section. And I think what @garciasolero meant above is that the quoted text should also be added to the CIBA section in https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#authentication-request.

Reviewer 2: This would fix the part of issue #283 below:
> In addition, we should apply the same maximum request object lifetime policy in both CIBA and Authorization Code flows that we require in the [client assertion](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#client-authentication) for authentication.

Reviewer: In the current profile included in this pull request, there are only two sections addressing the JWT lifetime limited to 300 seconds: 

1. in the paragraph added in this PR for the request parameter within the auth code flow
2. in the section of the client assertion for authentication that already existed in previous versions of the profile. 

There is no mention of the JWT lifetime for the request parameter in the CIBA section. If we limit it for those cases, we should also limit it for CIBA as proposed in the issue.

>   * The request SHALL be rejected by the authorisation server if the exp claim is more than 300 seconds later than the time of receipt. Additionally, if the iat claim is present, the request SHALL be rejected if the difference between the exp claim and iat claim is more than 300 seconds.

Reviewer 3: > if the `iat` claim is present

But the `iat` claim MUST be present, because it is mandatory. So the text can be simplified.

```suggestion
  * The request SHALL be rejected by the authorisation server if the iat claim is later than the time of receipt, or if the difference between the exp claim and iat claim is greater than 300 seconds.
```

Reviewer 2: I've committed this change to merge the PR, as it has already been considered in other related discussions.
